openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -95,8 +115,14 @@ def extract_dates_from_course(course):
             for _, section, weeks_to_complete in spaced_out_sections(course):
                 section_date_items = []
                 for subsection in section.get_children():
-                    section_date_items.extend(_gather_graded_items(subsection, weeks_to_complete))
-
+                    # If custom pacing is set on a subsection, apply the set relative
+                    # date to all the content inside the subsection. Otherwise
+                    # apply the current PLS logic for self paced courses.
+                    due_num_weeks = subsection.fields['due_num_weeks'].read_from(subsection)
+                    if (CUSTOM_PLS.is_enabled(course.id) and due_num_weeks):
+                        section_date_items.extend(_get_custom_pacing_children(subsection, due_num_weeks))
+                    else:
+                        section_date_items.extend(_gather_graded_items(subsection, weeks_to_complete))
                 if section_date_items and section.graded:
                     date_items.append((section.location, weeks_to_complete))
```

Comment:
How do these play with the custom PLS?

Reply:
thank you for pointing this out! Just realized that this goes against my previous logic where you can set due dates for non graded sections too.